SoomlaiOSProfile/SoomlaProfile.m

```diff
@@ -100,9 +104,14 @@ - (void)loginWithProvider:(Provider)provider andPayload:(NSString *)payload andR
         [authController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
     }
     @catch (NSException *exception) {
-
-        // TODO: implement logic like in java that will raise the exception. Currently not raised
-        [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        @try {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [socialController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
+        }
+        @catch (NSException *exception) {
+            // TODO: implement logic like in java that will raise the exception. Currently not raised
+            [gameServicesController loginWithProvider:provider andAutoLogin:NO andPayload:payload andReward:reward];
```

Comment:
@EugeneButusov @vedi Let's add a separate task for improving the design here, it's not good anyway.